crates/ruff_linter/src/rules/pyupgrade/rules/pep695/non_pep695_generic_class.rs

```diff
@@ -118,9 +121,9 @@ pub(crate) fn non_pep695_generic_class(checker: &mut Checker, class_def: &StmtCl
         return;
     };
 
-    // TODO(brent) only accept a single, Generic argument for now. I think it should be fine to have
-    // other arguments, but this simplifies the fix just to delete the argument list for now
-    let [Expr::Subscript(ExprSubscript {
+    // only handle the case where Generic is at the end of the argument list, in line with PYI059
+    // (generic-not-last-base-class)
```

Comment:
```suggestion
    // (generic-not-last-base-class). If it comes elsewhere, it results in a runtime error.
    // In stubs it's not *strictly* necessary for `Generic` to come last in the bases tuple,
    // but it would cause more complication for us to handle stubs specially,
    // and probably isn't worth the bother.
```

Could we also add a test for the case where `Generic` isn't last in the bases tuple? I'm particularly interested in whether we issue a diagnostic for classes where `Generic` is not last in the bases tuple. For classes like that, I think we should probably issue a diagnostic, but not a fix...?

Reply:
Ah, I was hoping we could add something like `tail_classes @ ..` to the end of the slice and magically handle `Generic` not being at the end, but you can only do `..` once in a slice pattern. We currently bail out too early to give a diagnostic here, but I'll try to fix that!

Would it be bad to offer a fix anyway? I think most of the effort will go into finding `Generic`; the fix logic should be mostly the same, if not identical.

Reply:
> Would it be bad to offer a fix anyway?

I think so, because this is a stylistic rule, but if we offered a fix when `Generic` is in the middle of the bases list we'd be changing the behaviour of the program :/ we'd be turning broken code into code that would no longer raise an exception.

Now, that might be something that users would thank us for, of course 😄 but it still doesn't feel like something a _stylistic_ rule should be doing. If this rule's primary purpose was to complain about _possibly incorrect_ code, that would be a different question.